R/weighted_mean_median_sd_mad.R

```diff
@@ -102,3 +104,27 @@ weighted_mad <- function(x, weights = NULL, constant = 1.4826, verbose = TRUE, .
 
   pos
 }
+
+.clean_missings <- function(x, weights) {
+  flag <- FALSE
+  if (any(is.infinite(x)) || any(is.infinite(weights))) {
+    # remove Inf
+    x[is.infinite(x)] <- NA
+    weights[is.infinite(weights)] <- NA
+    flag <- TRUE
+  }
+
+  if (anyNA(x) || anyNA(weights)) {
+    # remove missings
+    x[is.na(weights)] <- NA
+    weights[is.na(x)] <- NA
```

Comment:
These two lines are not necessary since they're already `NA`

Reply:
You may have different observations with `NA` in `x` and `weights`. This ensures something like `complete.cases()` for `x` and `weights` when we use `na.omit()`.

Reply:
ah right, I didn't notice `x` and `weights` were inverted